src/main/java/randoop/generation/DemandDrivenInputCreation.java

```diff
@@ -0,0 +1,489 @@
+   * @param onlyReceivers if true, only return sequences that are appropriate to use as a method
+   *     call receiver
+   * @return method sequences that produce objects of the required type
+   */
+  public static SimpleList<Sequence> createInputForType(
+      SequenceCollection sequenceCollection,
+      Type t,
+      boolean exactTypeMatch,
+      boolean onlyReceivers) {
+
+    EXACT_TYPE_MATCH = exactTypeMatch;
+    ONLY_RECEIVERS = onlyReceivers;
+
+    // All constructors/methods found that return the demanded type.
+    Set<TypedOperation> producerMethods = getProducerMethods(t);
+
+    // For each producer method, create a sequence that produces an object of the demanded type
+    // if possible, or produce a sequence that leads to the eventual creation of the demanded type.
+    for (TypedOperation producerMethod : producerMethods) {
+      Sequence newSequence = getInputAndGenSeq(sequenceCollection, producerMethod);
+      if (newSequence != null) {
+        // Execute the sequence and store the resultant sequence in the sequenceCollection
+        // if the execution is successful.
+        executeAndAddToPool(sequenceCollection, Collections.singleton(newSequence));
+      }
+    }
+
+    // Get all method sequences that produce objects of the demanded type from the
+    // sequenceCollection.
+    SimpleList<Sequence> result = getCandidateMethodSequences(sequenceCollection, t);
+
+    if (GenInputsAbstract.demand_driven_logging != null) {
+      logUnspecifiedClasses();
+    }
+
+    return result;
+  }
+
+  /**
+   * Returns a set of methods with a given return type.
+   *
+   * <p>The method checks for all visible methods and constructors in the specified type that return
+   * the same type. It also recursively searches for inputs needed to execute a method that returns
+   * the type. The recursive search terminates if the current type is a primitive type or if it has
+   * already been processed.
+   *
+   * @param t the return type of the resulting methods
+   * @return a set of TypedOperations that construct objects of the specified type t
+   */
+  public static Set<TypedOperation> getProducerMethods(Type t) {
+    // The set of producer methods that construct objects of the specified type.
+    Set<TypedOperation> producerMethods = new LinkedHashSet<>();
+
+    for (String className : SPECIFIED_CLASSES) {
+      try {
+        Class<?> cls = Class.forName(className);
+        Type specifiedType = new NonParameterizedType(cls);
+        producerMethods.addAll(iterativeProducerMethodSearch(t, specifiedType));
+      } catch (ClassNotFoundException e) {
+        throw new RandoopBug("Class not found: " + className);
+      }
+    }
+
+    // Recursively search for methods that construct objects of the specified type.
+    producerMethods.addAll(iterativeProducerMethodSearch(t, t));
+
+    return producerMethods;
+  }
+
+  /**
+   * Helper method for getProducerMethods. This method recursively searches for methods that
```

Comment:
In what sense is this method recursive?
Also, why does it need to be recursive?  Wouldn't it be enough to search in all the known types (and perhaps in all the types used by them)?
Can the set of all known types (which is a superset, closed under formal parameters, of the user-specified types) be computed once early in execution and just used thereafter, rather than being recursively discovered multiple times?